keepalived/vrrp/vrrp_print.c

```diff
@@ -418,3 +421,182 @@ vrrp_print_stats(void)
+			json_object_new_int(vrrp->vrid));
+		json_object_object_add(json_data, "base_priority",
+			json_object_new_int(vrrp->base_priority));
+		json_object_object_add(json_data, "effective_priority",
+			json_object_new_int(vrrp->effective_priority));
+		json_object_object_add(json_data, "vipset",
+			json_object_new_boolean(vrrp->vipset));
+
+		//Virtual IPs are stored in a list
+		for (f = LIST_HEAD(v); f; ELEMENT_NEXT(f)) {
+			ip_address_t *vip = ELEMENT_DATA(f);
+			char ipaddr[INET6_ADDRSTRLEN];
+			inet_ntop(vrrp->family, &(vip->u.sin.sin_addr.s_addr),
+				ipaddr, INET6_ADDRSTRLEN);
+			json_object_array_add(vips,
+				json_object_new_string(ipaddr));
+		}
+		json_object_object_add(json_data, "vips", vips);
+
+		json_object_object_add(json_data, "promote_secondaries",
+			json_object_new_boolean(vrrp->promote_secondaries));
+		json_object_object_add(json_data, "evip_add_ipv6",
+			json_object_new_boolean(vrrp->evip_add_ipv6));
+		json_object_object_add(json_data, "master_adver_int",
+			json_object_new_int(vrrp->master_adver_int));
+		json_object_object_add(json_data, "accept",
+			json_object_new_int(vrrp->accept));
+		json_object_object_add(json_data, "iptable_rules_set",
+			json_object_new_boolean(vrrp->iptable_rules_set));
+		json_object_object_add(json_data, "nopreempt",
+			json_object_new_boolean(vrrp->nopreempt));
+		json_object_object_add(json_data, "preempt_delay",
+			json_object_new_int(vrrp->preempt_delay));
+		json_object_object_add(json_data, "preempt_time",
+			json_object_new_int(vrrp->preempt_time.tv_sec));
+		json_object_object_add(json_data, "state",
+			json_object_new_int(vrrp->state));
+		json_object_object_add(json_data, "init_state",
+			json_object_new_int(vrrp->init_state));
+		json_object_object_add(json_data, "wantstate",
+			json_object_new_int(vrrp->wantstate));
+		json_object_object_add(json_data, "fd_in",
+			json_object_new_int(vrrp->fd_in));
+		json_object_object_add(json_data, "fd_out",
+			json_object_new_int(vrrp->fd_out));
+		json_object_object_add(json_data, "debug",
+			json_object_new_int(vrrp->debug));
+		json_object_object_add(json_data, "quick_sync",
+			json_object_new_int(vrrp->quick_sync));
+		json_object_object_add(json_data, "version",
+			json_object_new_int(vrrp->version));
+		json_object_object_add(json_data, "smtp_alert",
+			json_object_new_boolean(vrrp->smtp_alert));
+		json_object_object_add(json_data, "notify_exec",
+			json_object_new_boolean(vrrp->notify_exec));
+		json_object_object_add(json_data, "ms_down_timer",
+			json_object_new_int(vrrp->ms_down_timer));
+		json_object_object_add(json_data, "sands",
+			json_object_new_int(vrrp->sands.tv_sec));
+		json_object_object_add(json_data, "auth_type",
+			json_object_new_int(vrrp->auth_type));
+		json_object_object_add(json_data, "ip_id",
+			json_object_new_int(vrrp->ip_id));
+
+		// Dump stats to json
+		json_object_object_add(json_stats, "advert_rcvd",
+			json_object_new_int64(vrrp->stats->advert_rcvd));
```

Comment:
A number of the stats values are uint64_t and need casting to int64_t to work around the limitations of json-c.